build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -185,7 +185,7 @@ class _SingleBuild {
       _buildPhases = buildImpl._buildPhases,
       _buildPhasePool = List.generate(
         buildImpl._buildPhases.length,
-        (_) => Pool(buildPhasePoolSize),
+        (_) => Pool(1),
```

Comment:
does a pool of 1 make sense?

Reply:
Good question! Looking at where the pool is used, there are no concurrent calls possible any more. So, removed. Thanks.